core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -816,7 +816,7 @@ public void tryReconnect() {
         return new Slave.JnlpJar(fileName);
     }
 
-    @WebMethod(name="slave-agent.jnlp")
+    @WebMethod(name="slave-agent.jnlp") // backward compatibility
```

Comment:
Maybe log something here, i assume it won't be backward compatible forever?

Reply:
Would keep this around at least for a few years, as it's not expensive to support additional URLs. I added this comment to prevent people from search & replacing this accidentally.